From f7b125ac79824b9c47111a4ba1928369dc9435ee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?= <avarab@gmail.com>
Date: Sat, 24 Jul 2010 19:58:37 +0000
Subject: [PATCH v2 4/7] Makefile: Add coverage-report-cover-db target
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add a target to convert the *.gcov files to a Devel::Cover
database. That database can subsequently be formatted by the cover(1)
tool, which is included with Devel::Cover.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 .gitignore |    1 +
 Makefile   |    4 ++++
 2 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/.gitignore b/.gitignore
index 0e61845..bd53c02 100644
--- a/.gitignore
+++ b/.gitignore
@@ -208,3 +208,4 @@
 *.gcno
 *.gcov
 /coverage-untested-functions
+/cover_db
diff --git a/Makefile b/Makefile
index e9f8ef8..f2c680d 100644
--- a/Makefile
+++ b/Makefile
@@ -2303,3 +2303,7 @@ coverage-untested-functions: coverage-report
 	grep '^function.*called 0 ' *.c.gcov \
 		| sed -e 's/\([^:]*\)\.gcov: *function \([^ ]*\) called.*/\1: \2/' \
 		| tee coverage-untested-functions
+
+coverage-report-cover-db: coverage-report
+	gcov2perl -db cover_db *.gcov
+
-- 
1.7.0.4

